sdk/core/azure-core/azure/core/settings.py

```diff
@@ -49,7 +53,7 @@
 __all__ = ("settings",)
 
 
-class _Unset(object):
+class _Unset(AbstractSpan):
```

Comment:
so for logging, the value could be unset which does not necessarily inherit from abstract span. If later the settings were to store a different object that was not a span, it would throw an error.

Reply:
Not sure I get your commend, since this object is a sentinel that you check with the "is" keyword later, so you shall never return this object?
It will make the type more complicated, but guess it might bring clarity since yourself were confused :)